### PR TITLE
feat: better history

### LINKED ISSUE #85: Refactor transaction history to handle more than Send
We need to handle
- Mint/Burn
- Payout/BurnHeldBalance
- Submit
- Execution of messages contained in a passing proposal

We also need to handle the case where a proposal contains multiple messages.

### PR DESCRIPTION
**Pull request by fmorency**
Include and differentiate Send, Recieve, Mint, Burn, Payout, BurnHeldBalance and executed Proposals transaction types in transaction history.

This PR adds the following environment variables

```
NEXT_PUBLIC_TESTNET_EXPLORER_URL : Testnet block explorer URL
NEXT_PUBLIC_MAINNET_EXPLORER_URL : Mainnet block explorer URL
NEXT_PUBLIC_TESTNET_INDEXER_URL : Testnet indexer URL
NEXT_PUBLIC_MAINNET_INDEXER_URL : Mainnet indexer URL
```

The new variables were added to the existing CI environments.

This PR relies on the endpoint selector for the block explorer and indexer URLs.
This PR adds the block explorer and indexer URL to the "Add endpoint" form.

Depends on https://github.com/liftedinit/yaci/pull/31

Fixes #85 

Tests are to be added in another PR...

## Release Notes

- **New Features**
	- Enhanced transaction handling with the addition of transaction types.
	- Dynamic endpoint management allows for flexible transaction and account link generation.

- **Bug Fixes**
	- Improved validation for indexer and explorer endpoints in the endpoint selector.

- **Refactor**
	- Streamlined test cases for the `HistoryBox` component by removing unnecessary props.
	- Updated transaction fetching logic to improve clarity and maintainability.

- **Chores**
	- Updated mock transaction data to include transaction types for better categorization.

### CODE REVIEW

File: components/bank/components/historyBox.tsx [hunk 1]
```diff
@@ -91,6 +95,71 @@
...
+    } else if (tx.data.tx_type === HistoryTxType.MINT || tx.data.tx_type === HistoryTxType.PAYOUT) {
+      return 'text-green-500';
+    } else if (
+      tx.data.tx_type === HistoryTxType.BURN ||
+      tx.data.tx_type === HistoryTxType.BURN_HELD_BALANCE
+    ) {
+      return 'text-red-500';
+    }
+    return null;
+  }
```
[the remark this answers is not in the document]
Comment (fmorency) [reply components/bank/components/historyBox.tsx:161]: Create a new ticket and create those unit tests.

File: hooks/useQueries.ts [hunk 2]
```diff
@@ -750,68 +752,147 @@
...
+    case `/lifted.init.manifest.v1.MsgBurnHeldBalance`:
+      return {
+        data: {
+          tx_type: HistoryTxType.BURN_HELD_BALANCE,
+          from_address: message.authority,
+          to_address: message.authority,
+          amount: message.burnCoins,
+        },
+      };
+    default:
```
[the remark this answers is not in the document]
Comment (fmorency) [reply hooks/useQueries.ts:836]: Create a new ticket and create those unit tests.

File: hooks/useQueries.ts [hunk 3]
```diff
@@ -750,68 +752,147 @@
...
+    if (formattedMessage) {
+      messages.push({
+        tx_hash: tx.id,
+        block_number: parseInt(tx.data.txResponse.height),
+        formatted_date: tx.data.txResponse.timestamp,
+        ...formattedMessage,
+      });
+    }
+  }
+  return messages;
```
[the remark this answers is not in the document]
Comment (fmorency) [reply hooks/useQueries.ts:874]: Create a new ticket and create those unit tests.

File: tests/mock.ts [hunk 4]
```diff
@@ -200,6 +200,7 @@ export const mockTransactions: TransactionGroup[] = [
     block_number: 1,
     formatted_date: '2023-05-01T12:00:00Z',
     data: {
+      tx_type: HistoryTxType.SEND,
```
[the remark this answers is not in the document]
Comment (fmorency) [reply tests/mock.ts:203]: Create a new ticket and create those unit tests.

File: components/react/endpointSelector.tsx [hunk 5]
```diff
@@ -78,6 +78,57 @@ const validateAPIEndpoint = async (url: string) => {
...
+    //   return false;
+    // }
+    //
+    // return response.ok;
+    return true;
+  } catch (error) {
+    console.error('Explorer Validation error:', error);
+    return false;
+  }
+};
```
[the remark this answers is not in the document]
Comment (fmorency) [reply components/react/endpointSelector.tsx:130]: Had CORS issue locally.

**Review by jgryffindor — APPROVED** [review]
utACK

**Comment by fmorency** [comment]
> looks good to me. Some of the code rabbit recommendations are okay but I think most are pretty dumb for this.

Yeah I have mixed feelings about this CodeRabbit tool...

File: hooks/useQueries.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (fmorency) [reply hooks/useQueries.ts:836]: Write the unit tests and add them to the issue.

File: tests/mock.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (fmorency) [reply tests/mock.ts:203]: Write the unit tests and add them to the issue.

**Pull request merged**